providers/os/resources/purl/purl.go

```diff
@@ -77,9 +77,14 @@ func NewPackageURL(pf *inventory.Platform, t Type, name, version string, modifie
 		// use the platform architecture for the package
 		purl.Arch = pf.Arch
 
-		if pf.Name != "" {
-			purl.Namespace = pf.Name
+		platformName := pf.Name
```

Comment:
I recommend we rename it to `purlNamespace := pf.Name` to make clear that we do not change the platform name but instead determine the correct purl namespace.